tasks/sanitize-checks.yml

```
@@ -0,0 +1,9 @@
+- name: Defend against defined but null datadog_checks variable
+  set_fact:
+      datadog_checks: "{{datadog_checks | default({})}}"
+  when: datadog_checks is none
```

I'd also guard against empty strings and other oddities by doing this:
```suggestion
  set_fact:
      datadog_checks: "{{ datadog_checks | default({}, true) }}"
```

---

Excellent, I was not aware of this behaviour regarding true.   Good stuff, more concise.